executor/include/execution/trait/can_query.hpp

```diff
@@ -0,0 +1,18 @@
+//
+// Created by shrijit on 6/20/20.
+//
+
+#pragma once
+
+#include <execution/trait/common_traits.hpp>
+#include <type_traits>
+
+namespace execution {
+template <typename Executor, typename Property,
+          typename std::enable_if_t<
+              Property::template is_applicable_property_v<Executor>, int> = 0>
+constexpr auto query(Executor &&ex, const Property &p) noexcept {
+  return Property::template static_query<
+      std::remove_cv_t<std::remove_reference_t<Executor>>>();
```

Comment:
I'm still convinced you don't need `Property::static_query` to exist to make this check. You can make the call directly on the executors query methods.

Reply:
`static_query` is part of the [proposal](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1393r0.html)

Reply:
This should answer the your first question in https://github.com/shrijitsingh99/unified-executors/pull/7#issuecomment-652589894